aspnetcore/fundamentals/servers/snippets/10.x/memory-pool-factory.cs

```diff
@@ -0,0 +1,13 @@
+services.AddSingleton<IMemoryPoolFactory<byte>,
+CustomMemoryPoolFactory>();
+
+public class CustomMemoryPoolFactory : IMemoryPoolFactory<byte>
+{
+    public MemoryPool<byte> Create()
+    {
+        // Return a custom MemoryPool implementation
+        // or the default, as is shown here.
+        return MemoryPool<byte>.Shared;
+    }
+}
+
```

Comment:
```suggestion
services.AddSingleton<IMemoryPoolFactory<byte>,
CustomMemoryPoolFactory>();

public class CustomMemoryPoolFactory : IMemoryPoolFactory<byte>
{
    public MemoryPool<byte> Create()
    {
        // Return a custom MemoryPool implementation
        // or the default, as is shown here.
        return MemoryPool<byte>.Shared;
    }
}

```

When you're using a memory pool, it's often a good idea to check the pool's [MaxBufferSize](https://github.com/dotnet/api/system.buffers.memorypool-1.maxbuffersize?view=net-9.0#system-buffers-memorypool-1-maxbuffersize) property.